packages/react-components/lib/tasks/create-task.tsx

```diff
@@ -1345,69 +1422,78 @@ export function CreateTaskForm({
   // no memo because deps would likely change
   const handleSubmit = async (scheduling: boolean) => {
     if (!submitTasks) {
-      onSuccess && onSuccess(taskRequests);
+      onSuccess && onSuccess([taskRequest]);
       return;
     }
 
     const requester = scheduling ? `${user}__scheduled` : user;
 
-    for (const t of taskRequests) {
-      t.requester = requester;
-      t.unix_millis_request_time = Date.now();
-
-      // Workaround where all the task category need to be compose.
-      if (t.category !== 'patrol') {
-        t.category = 'compose';
-
-        const goToOneOfThePlaces: GoToOneOfThePlacesActivity = {
-          category: 'go_to_place',
-          description: {
-            one_of: emergencyLots.map((placeName) => {
-              return {
-                waypoint: placeName,
-              };
-            }),
-            constraints: [
-              {
-                category: 'prefer_same_map',
-                description: '',
-              },
-            ],
-          },
-        };
-        const deliveryDropoff: DropoffActivity = {
-          category: 'perform_action',
-          description: {
-            unix_millis_action_duration_estimate: 60000,
-            category: 'delivery_dropoff',
-            description: {},
-          },
-        };
-        const onCancelDropoff: OnCancelDropoff = {
-          category: 'sequence',
-          description: [goToOneOfThePlaces, deliveryDropoff],
-        };
-        taskRequest.description.phases[1].on_cancel = [onCancelDropoff];
+    const request = { ...taskRequest };
+    request.requester = requester;
+    request.unix_millis_request_time = Date.now();
+
+    if (
+      taskType === 'delivery_pickup' ||
+      taskType === 'delivery_sequential_lot_pickup' ||
+      taskType === 'delivery_area_pickup'
+    ) {
+      const goToOneOfThePlaces: GoToOneOfThePlacesActivity = {
+        category: 'go_to_place',
+        description: {
+          one_of: emergencyLots.map((placeName) => {
+            return {
+              waypoint: placeName,
+            };
+          }),
+          constraints: [
+            {
+              category: 'prefer_same_map',
+              description: '',
+            },
+          ],
+        },
+      };
+      const deliveryDropoff: DropoffActivity = {
+        category: 'perform_action',
+        description: {
+          unix_millis_action_duration_estimate: 60000,
```

Comment:
Not a blocker since it's the same as the previous code, but shouldn't this be set by rmf?

Reply:
No, it's meant to be set by the user, estimating how long each perform action is supposed to take to help RMF plan. It was set manually here, because it is thankfully an invariant (for picking up or dropping off carts in the current project).

But it was also a feature request that we don't bog down the users with too many inputs to fill in, so this is the estimate we chose to use statically.

Reply:
Add another `FIXME` to this, this design isn't good as we are making assumptions on the deployment again.

Reply:
https://github.com/open-rmf/rmf-web/pull/903/commits/03aa973ca9a849be52651fc6a82c4b252a41cdfd

When I introduce back the demos task requests into this, I will refactor them similarly to https://github.com/open-rmf/rmf-web/pull/893/files#diff-70a086cc44648bbf06fb810e18f59a99cfcf139952b6eda2fe1447b0faf31a7c, where they are easier to maintain, testable and have better separation of concern